lib/util/stream_utils.js

```diff
@@ -263,16 +263,31 @@ shaka.util.StreamUtils = class {
     // |video.width| and |video.height| can be undefined, which breaks
     // the math, so make sure they are there first.
     if (video && video.width && video.height) {
-      if (!inRange(video.width,
-          restrictions.minWidth,
-          Math.min(restrictions.maxWidth, maxHwRes.width))) {
-        return false;
-      }
+      if (video.width >= video.height) {
```

Comment:
I think this might be clearer if you made local variables for video width and video height, and then had only two checks.
E.g.

```js
let videoWidth = video.width;
let videoHeight = video.height;
if (videoHeight > videoWidth) {
  // Vertical video.
  [videoWidth, videoHeight] = [videoHeight, videoWidth];
}
if (!inRange(videoWidth, restrictions.minWidth, Math.min(restrictions.maxWidth, maxHwRes.width))) {
  return false;
}
if (!inRange(videoHeight, restrictions.minHeight, Math.min(restrictions.maxHeight, maxHwRes.height))) {
  return false;
}
```